### PR TITLE
Integrate flatPickr

### LINKED ISSUE #2011: Integrate `flatPickr`
We need our players to enter date-time values for new orders.

JSON-Editor supports the `flatpickr` component, but it's not working in our templates.

The text is locked, as-if JSON-Editor wants to make users user the control, but the control doesn't appear.

screencast

There is a story supporting this here: http://localhost:6611/?path=/story/local-form-elements-newmessage--just-land-activity (note: right now I haven't pushed that story yet, but will soon)

### PR DESCRIPTION
**Pull request by lebaphi**
Fixes #2011

### CODE REVIEW

**Comment by lebaphi** [comment]
I've updated json-editor and flatpickr to the latest version.

https://user-images.githubusercontent.com/26705621/200867265-cf282eda-a86b-4da3-95e1-819cb437a1f8.mov

File: packages/client/public/index.html [hunk 1]
```diff
@@ -7,6 +7,7 @@
   <link rel='shortcut icon' href='%PUBLIC_URL%/favicon.ico' />
   <link rel='stylesheet' href='%PUBLIC_URL%/leaflet.css' />
   <link rel='stylesheet' href='%PUBLIC_URL%/leaflet.select/leaflet.control.select.css' />
+  <link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/flatpickr@4.6.3/dist/flatpickr.min.css">
```
Comment (IanMayo) [inline packages/client/public/index.html]: we'll need stylesheets to be stored in the repo please, we won't be able to load them at run-time.

File: packages/components/.storybook/preview-head.html [hunk 2]
```diff
@@ -1,3 +1,4 @@
 <link href='https://fonts.googleapis.com/css?family=Open Sans' rel='stylesheet'>
 <link rel="stylesheet" href="https://fonts.googleapis.com/icon?family=Material+Icons" />
-<link rel='stylesheet' href='%PUBLIC_URL%/leaflet.select/leaflet.control.select.css' />
\ No newline at end of file
+<link rel='stylesheet' href='%PUBLIC_URL%/leaflet.select/leaflet.control.select.css' />
+<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/flatpickr@4.6.3/dist/flatpickr.min.css">
```
Comment (IanMayo) [inline packages/components/.storybook/preview-head.html]: this will also need to be stored in the repo too, please.

Comment (lebaphi) [reply packages/components/.storybook/preview-head.html]: okay @IanMayo

**Comment by IanMayo** [comment]
Thanks for this @lebaphi 

It's working in my browser. But, we're not getting date *and* time.   It's just date.  The `format` is specified as `datetime-local`, which is what `json-editor` says should be used.

I'm expecting an editor more like this:
image

Any ideas?

**Comment by lebaphi** [comment]
let me check if the flatpickr provides some options to do that thing, but in tomorrow 😄

**Comment by IanMayo** [comment]
Thanks @philb89 - but we really need the time component to work, too.  For the `start date` and `end date` - the players need to enter date and time (well, just the date and hour, actually).

Is it worth experimenting with version `2.8` of `json-editor`?

**Comment by lebaphi** [comment]
> Thanks @philb89 - but we really need the time component to work, too. For the `start date` and `end date` - the players need to enter date and time (well, just the date and hour, actually).
> 
> Is it worth experimenting with version `2.8` of `json-editor`?

let me try with the old version 1.3.5

**Comment by lebaphi** [comment]
I reverted json editor back to version 1.3.5 but got the same issue 🤔 

https://user-images.githubusercontent.com/26705621/201139603-50163517-81a8-42b9-bb1d-a317b1ded56c.mov

**Comment by lebaphi** [comment]
https://user-images.githubusercontent.com/26705621/201507725-de97f481-0994-49ea-b118-a729f1d15a69.mov

**Comment by IanMayo** [comment]
I've brought this PR up to date, but it doesn't seem to be working @lebaphi 

Would you mind updating it please?

**Comment by lebaphi** [comment]
sure Ian, Im will update in the next 2hrs

**Comment by lebaphi** [comment]
Hi @IanMayo , I checked the schema

image

I think this issue cause we lost the option `time_24hr: true` in the flatpickr options, and the format should be `datetime-local`

image

**Comment by IanMayo** [comment]
Hello @lebaphi - I'm trying to get this sorted.

It looks like json-editor is ignoring the template values for flatpickr options.

I'm using this story:
http://localhost:6611/?path=/story/local-molecules-jsoneditor--planning-message

The template being used in the story is this one:
https://github.com/serge-web/serge/pull/2027/files#diff-e1f45060503678cae7229e92041f24a3fe484b7444b2aac772735f0402d57238L13780

Note that `endDate` has flatpickr properties, but `startDate` & `endDate` both seem to have the same date-time control in the story

**Comment by IanMayo** [comment]
Hello @lebaphi - this could be _another_ issue that could magically go away with an update to `json-editor`.

**Comment by lebaphi** [comment]
> Hi @lebaphi - this could be an another issue that can magically disappear when `json-editor` is updated.

I don't think so @IanMayo, I upgraded json editor to latest version from the beginning of this PR but it didn't work, then I reverted back to 1.3.5 but had the same problem. https://github.com/serge-web/serge/pull/2027#issuecomment-1310487274
I think that means we still need to refer the solution on this PR.

**Comment by IanMayo** [comment]
Hello @lebaphi - could you have a go at fixing the conflicts here please?

I had a go - but got stuck :-(

**Comment by lebaphi** [comment]
> Hello @lebaphi - could you have a go at fixing the conflicts here please?
> 
> 
> 
> I had a go - but got stuck :-(

sure @IanMayo but tomorrow 😂

**Comment by lebaphi** [comment]
Wait for me to test again..

**Comment by lebaphi** [comment]
Hi @IanMayo, I tested on the json-editor v2.8.0

Screen Shot 2022-12-30 at 2 34 58 PM

**Pull request merged**